src/main/java/org/wise/portal/presentation/web/controllers/tag/TagProjectController.java

```diff
@@ -0,0 +1,88 @@
+package org.wise.portal.presentation.web.controllers.tag;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.wise.portal.dao.ObjectNotFoundException;
+import org.wise.portal.domain.project.Project;
+import org.wise.portal.domain.user.User;
+import org.wise.portal.domain.usertag.UserTag;
+import org.wise.portal.presentation.web.response.ResponseEntityGenerator;
+import org.wise.portal.service.project.ProjectService;
+import org.wise.portal.service.user.UserService;
+import org.wise.portal.service.usertags.UserTagsService;
+
+@RestController
+@Secured({ "ROLE_TEACHER" })
+@RequestMapping("/api")
+public class TagProjectController {
+
+  @Autowired
+  private ProjectService projectService;
+
+  @Autowired
+  private UserService userService;
+
+  @Autowired
+  private UserTagsService userTagsService;
+
+  @PutMapping("/projects/tag/{tagId}")
+  protected ResponseEntity<List<Map<String, Object>>> addTagToProjects(Authentication auth,
+      @RequestBody List<Long> projectIds, @PathVariable("tagId") Long tagId) throws Exception {
+    User user = userService.retrieveUserByUsername(auth.getName());
+    UserTag userTag = userTagsService.get(tagId);
+    List<Project> projects = getProjects(projectIds);
+    for (Project project : projects) {
+      userTagsService.applyTag(project, userTag);
+    }
+    return ResponseEntityGenerator.createSuccess(createProjectsResponse(user, projects));
```

Comment:
Can this function just return a ```List...``` instead of ```ResponseEntity<List...>```?
```suggestion
    return createProjectsResponse(user, projects);
```